fedn/fedn/clients/reducer/control.py

```diff
@@ -1,10 +1,62 @@
 from .state import ReducerState
+import copy
+
+
+import fedn.common.net.grpc.fedn_pb2 as fedn
+import fedn.common.net.grpc.fedn_pb2_grpc as rpc
+import grpc
+
+from fedn.algo.fedavg import FEDAVGCombiner
 
 class ReducerControl:
```

Comment:
This class got fat pretty quick, perhaps this should be moved to a nother layer of indirection to allow for alternate control strategies, hence moving to a ReducerControlStrategy

Reply:
I think those imports are not actually used here, removing them. 

Yes, I think this should become abstract, hooking in implementations of "spread_model" and "resolve" (those names should be worked on as well). In the general case, these can / will also be completely decentralized protocols.